src/commands/compile.ts

```diff
@@ -11,11 +11,18 @@ interface CompileArgv extends CommonArgv {
 
 export async function compile(
   settings: Settings,
-  content: string,
+  rawContents: string,
+  filename: string | null = null,
   shadow = false,
 ): Promise<string> {
   const parsedSettings = await parseSettings(settings, shadow);
-  return compilePlaceholders(parsedSettings, content, shadow);
+  const contents = await compileIncludes(
+    parsedSettings,
+    rawContents,
+    filename ? new Set([filename]) : new Set(),
```

Comment:
Hmmm... This could allow including the same file twice because the path names may not match up. We should make it so that we resolve all paths to absolute paths.